blueocean-dashboard/src/main/js/components/PullRequests.jsx

```diff
@@ -70,27 +79,32 @@ export class PullRequests extends Component {
             }
 
             config.pipeline = pipelineName;
-            this.props.fetchBranchesIfNeeded(config);
         }
     }
 
     render() {
-        const { branches } = this.props;
-
         if (this.state.unsupportedJob) {
             return (<NotSupported />);
         }
 
+        const { pullRequests } = this.props;
+
         // early out
-        if (!branches) {
+        if (!pullRequests) {
```

Comment:
can't really follow the changes here as it seems noisy. I guess eventually this will be capabilities based rather than data? (or not, as it may not have any branches but could?)